rclcpp_action/include/rclcpp_action/client.hpp

```diff
@@ -307,7 +310,11 @@ class Client : public ClientBase
    */
   std::shared_future<typename GoalHandle::SharedPtr>
   async_send_goal(
-    const Goal & goal, FeedbackCallback callback = nullptr, bool ignore_result = false)
+    const Goal & goal,
+    GoalResponseCallback goal_response_callback = nullptr,
+    FeedbackCallback feedback_callback = nullptr,
+    ResultCallback result_callback = nullptr,
+    bool ignore_result = false)
```

Comment:
@jacobperron (meta, can ignore): I know this is a purely subjective observation, biased by own prejudices, but since C++ lacks keyword arguments, I find having to pass multiple `nullptr` values to get to the one I actually care about somewhat annoying. Can we have some overloads for the most common use cases e.g. goal and feedback cb, goal, feedback cb and result cb, etc.?

Reply:
We've been going the way of "options structs" which alleviates some of this without expanding the number of signatures, also the signatures get confusing/ambiguous at some point.

But in general I agree that this becomes too fragile and inconvenient at some point.

Reply:
I could be persuaded to update this signature to accept a `SendGoalOptions` struct.

I was also thinking we might consider removing the `ignore_result` parameter and instead rely on checking if `result_callback` is set.

Reply:
Which ever is fine by me.

Reply:
I've added an options struct in f1c46f5. PTAL.

Reply:
Looks reasonable to me.